src/main/java/dataStructuresAndAlgorithms/challenges/fifoAnimalShelter/AnimalShelter.java

```diff
@@ -1,5 +1,66 @@
 package dataStructuresAndAlgorithms.challenges.fifoAnimalShelter;
 
 
+import dataStructuresAndAlgorithms.dataStructures.stacksAndQueues.Queue;
+
+import java.util.InputMismatchException;
+
 public class AnimalShelter {
+/********
+ * Instance variables
+ * */
+    Queue shelter;
+
+
+/********
+ * Constructors
+ * */
+    public AnimalShelter() { this.shelter = new Queue(); }
+
+
+/********
+ * Getters / Setters
+ * */
+    public Queue getShelter() { return this.shelter; }
+
+
+/********
+ * Instance Methods
+ * */
+
+    public void enqueue(Animal animal) {
+        try {
+            if (animal.getTypeOfAnimal() == "dog" || animal.getTypeOfAnimal() == "cat" ) {
+                shelter.enqueue(animal.getTypeOfAnimal());
+            } else {
+                throw new InputMismatchException("Please enter cat or dog");
+            }
+
+        } catch(Exception e) {
+
+            System.err.println("An error has occurred: " + e);
+        }
+    }
+
+
+    public Animal dequeue(Animal preference) {
+        try {
+            if (preference.getTypeOfAnimal() == "dog" || preference.getTypeOfAnimal() == "cat" ) {
+                while(shelter.peek() != preference) {
+                    shelter.enqueue(shelter.dequeue());
```

Comment:
While this will return the animal the user wants, but by dequeue-ing and enqueue-ing you've lost the order that the animals have arrived at the shelter